cli/src/cluster_query.rs

```diff
@@ -1339,20 +1335,16 @@ pub fn process_show_stakes(
     }
     let all_stake_accounts = rpc_client
         .get_program_accounts_with_config(&solana_stake_program::id(), program_accounts_config)?;
-    let stake_history_account = rpc_client.get_account(&stake_history::id())?;
     let clock_account = rpc_client.get_account(&sysvar::clock::id())?;
     let clock: Clock = from_account(&clock_account).ok_or_else(|| {
         CliError::RpcRequestError("Failed to deserialize clock sysvar".to_string())
     })?;
     progress_bar.finish_and_clear();
 
-    let stake_history = from_account(&stake_history_account).ok_or_else(|| {
-        CliError::RpcRequestError("Failed to deserialize stake history".to_string())
-    })?;
-
     let mut stake_accounts: Vec<CliKeyedStakeState> = vec![];
     for (stake_pubkey, stake_account) in all_stake_accounts {
         if let Ok(stake_state) = stake_account.state() {
+            let stake_activation = rpc_client.get_stake_activation(&stake_pubkey)?;
```

Comment:
Unfortunately, this doesn't quite work. If a stake account is undelegated, `getStakeActivation` returns a JSON-rpc error: `{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid param: stake account has not been delegated"},"id":1}`
...which errors out here and breaks the loop (or in the case of `solana stake-account`, prevents the account info being shown).

A simple (naive) fix might be to write this as:
```suggestion
            let stake_activation = rpc_client.get_stake_activation(&stake_pubkey).unwrap_or(
              RpcStakeActivation {
                state: StakeActivationState::Inactive,
                active: 0,
                inactive: 0,
              }
            );
```
although that would disguise other types of errors.